### PR TITLE
Use React Instead of Backbone

### PR DESCRIPTION
**Pull request by justinmc**
Issue: https://github.com/Autodesk/nbmolviz2d/issues/10

If we're going to buy into a framework with this project, let's make it a modern one we can use.  This branch worked nicely for dropping into the "workflow prototype" React project with very little effort.

The MDT project will gain support for this by wrapping the React component in Backbone in this PR: https://github.com/Autodesk/notebook-molecular-visualization/pull/22

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "presets": [
-    "es2015",
+    "latest",
```
Comment (maxbates) [inline .babelrc]: id recommend picking something static

File: example/js/example_settings.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+class ExampleSettings extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.onBlurSelection = this.onBlurSelection.bind(this);
```
Comment (maxbates) [inline example/js/example_settings.jsx]: https://babeljs.io/blog/2015/06/07/react-on-es6-plus -- look at the section on arrow functions to simplify this

File: src/utils/molecule_utils.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  /**
+   * Given two arrays of ids, return true if they contain the same values in any order,
+   * ignoring duplicates
+   * @param idsA {Array}
+   * @param idsB {Array}
+   * @returns {Boolean}
+   */
+  compareIds(idsA, idsB) {
+    // If one array is empty and the other isn't, they can't contain the same values
+    if ((!idsA.length && idsB.length) || (idsA.length && !idsB.length)) {
```
Comment (maxbates) [inline src/utils/molecule_utils.js]: if this is the only way you're using immutable... you might as well just Object.freeze() and ref equality check, or do a set yourself (its in ES6), or just use arrays:

```
function eqSet(a, b) {
    if (a.length !== b.length) return false;
    //or a map if these are big...
    for (var c of a) if (b.indexOf(a) < 0) return false;
    return true;
}
```

File: src/components/links.jsx [hunk 4]
```diff
@@ -40,50 +53,44 @@ const LinksView = Backbone.View.extend({
...
-
-    // keep edges pinned to their nodes
-    links.selectAll('line')
-      .attr('x1', d => d.source.x)
-      .attr('y1', d => d.source.y)
-      .attr('x2', d => d.target.x)
-      .attr('y2', d => d.target.y);
+  render() {
+    return (
+      <g className="links-container" ref={(c) => { this.linksContainer = c; }} />
```
Comment (maxbates) [inline src/components/links.jsx]: you should check if c is defined. sometimes its not (or at least in v14 i had this problem)

File: .babelrc (hunk 1 above)
Comment (justinmc) [inline .babelrc]: Good call

File: example/js/example_settings.jsx (hunk 2 above)
Comment (justinmc) [inline example/js/example_settings.jsx]: I was worried about class properties not being in the spec yet, but that really is much easier...  I just got it working using babel-eslint so I think I'll change all of these over.

File: src/components/links.jsx (hunk 4 above)
Comment (justinmc) [inline src/components/links.jsx]: Got it, I wasn't aware of that.

**Pull request merged**